### DISCUSSION TITLE
Enabling PII for ML capabilities.

### DISCUSSION THREAD

**Discussion by varsha-sl** [Sunbird-Ed/Community#50] [Feature request]
Use case for enabling PII :

With the idea of micro improvement and innovations campaign, it will become important for the Program manager/administrator to be able to identify end users so that they can be acknowledged, appreciated and awarded.

Similarly, to encourage end users to conduct school observations, it is important for the Program managers/administrators to know how many observations are done weekly, monthly and by who.

Open Questions :
 How will we ensure the Implementation on the API side? How will restrict and ensure that only Program Managers are accessing the PII data in the Program dashboard. Do we need to implement Access token in the data exhaust API to capture the user session to ensure safety of PII data access ?

Please find attached the solution note for the same.

[Solution note for PII and Discoverability of Programs.docx](https://github.com/Sunbird-Ed/Community/files/9148451/Solution.note.for.PII.and.Discoverability.of.Programs.docx)

https://docs.google.com/document/d/1E7po75zqpgdlvmN9zPE2M5pSlRVt3m9iRhP_DAtiF90/edit

Cc - [@aks30](https://github.com/aks30) [@vijiurs](https://github.com/vijiurs) [@kiranharidas187](https://github.com/kiranharidas187) [@varsha-sl](https://github.com/varsha-sl) [@Prateek-slokam](https://github.com/Prateek-slokam) [@aishwaryashikshalokam](https://github.com/aishwaryashikshalokam) [@kameshbhr](https://github.com/kameshbhr)

**Comment by varsha-sl** [comment]
[@kameshbhr](https://github.com/kameshbhr) can you please add your comments to the solution note and let me know if we need to add anyone else to this discusstion?

**Comment by varsha-sl** [comment]
@luckynvdu adding you to this disscusstion please take a look at the solution note

**Comment by aishwaryashikshalokam** [comment]
Dear @luckynvdu and @kameshbhr

We are awaiting your responses on the same so that we can take it forward. If we are good with the solution proposed, we can start working on the PRD and user flows. 

It's a high-priority item for us. Requesting both of you if we can get your go-ahead/suggestions/comments on the solution proposed.

**Reply by luckynvdu** [reply]
@varsha-sl The solution looks fine. Please go ahead and request to get it reviewed with legal team as well.

@aishwaryashikshalokam
